### PR TITLE
Components. Basic markup

### PR DESCRIPTION
**Pull request by melk0sha**
**Task 3. Components. Part 1**

Markup is done with React Components and React Fragments (parent-child).
Styles applied.
Used PropTypes.
Used <ErrorBoundary>component for catching and displaying errors.

### CODE REVIEW

File: src/components/resultsSection/movies/movie/movie.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        <MovieYear>{year}</MovieYear>
+      </MovieInfoWrapper>
+      <MovieGenres>{genresString}</MovieGenres>
+    </MovieWrapper>
+  );
+};
+
+Movie.propTypes = {
+  movie: object
+};
```
Comment (cate-blint) [inline src/components/resultsSection/movies/movie/movie.jsx]: Please extend propTypes for movie

File: src/components/resultsSection/movies/movies.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const MoviesCards = useMemo(() => movies.map((movie) => <Movie movie={movie} key={movie.id || Math.random()} />), [
+    movies
+  ]);
+
+  return <MoviesWrapper>{MoviesCards}</MoviesWrapper>;
+};
+
+Movies.propTypes = {
+  movies: array
+};
```
Comment (cate-blint) [inline src/components/resultsSection/movies/movies.jsx]: Please extend propTypes for movies

File: src/components/resultsSection/moviesSorting/moviesSorting.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {
+  SortingSpan,
+  MoviesSortingWrapper,
+  StyledDropdown
+} from "components/resultsSection/moviesSorting/moviesSorting.styled";
+import { SORT_BY_OPTIONS, SORT_BY_VALUES } from "consts";
```
Comment (cate-blint) [inline src/components/resultsSection/moviesSorting/moviesSorting.jsx]: It is better to use 'constants'

File: src/components/shared/actionMenu/actionMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+ActionMenu.propTypes = {
+  className: string,
+  options: arrayOf(
+    shape({
+      id: number,
+      value: string,
+      handler: func
+    })
+  )
```
Comment (cate-blint) [inline src/components/shared/actionMenu/actionMenu.jsx]: Please add propTypes for onOptionClick

File: src/components/shared/input/input.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { StyledInput } from "components/shared/input/input.styled";
+
+const Input = (props) => {
```
Comment (cate-blint) [inline src/components/shared/input/input.jsx]: What kind of props is passed here? Is it possible to add propTypes for these props?

File: src/api/__mock__/movies.json [hunk 6]
```diff
@@ -0,0 +1,84 @@
+{
+  "movies": [
+    {
+      "id": 0,
+      "name": "Movie 1",
+      "genres": [
+        { "id": 2, "name": "Comedy" },
+        { "id": 4, "name": "Crime" }
```
Comment (cate-blint) [inline src/api/__mock__/movies.json]: If you have genres values in genres.json, it is possible use only ids here for genres.

File: src/components/resultsSection/movies/movie/movie.jsx (hunk 1 above)
Comment (melk0sha) [reply src/components/resultsSection/movies/movie/movie.jsx]: Fixed

File: src/components/resultsSection/movies/movies.jsx (hunk 2 above)
Comment (melk0sha) [reply src/components/resultsSection/movies/movies.jsx]: Fixed

File: src/components/resultsSection/moviesSorting/moviesSorting.jsx (hunk 3 above)
Comment (melk0sha) [reply src/components/resultsSection/moviesSorting/moviesSorting.jsx]: We can't use 'constants' due to npm module name conflict. Something like that https://www.npmjs.com/package/constants

image

File: src/components/shared/actionMenu/actionMenu.jsx (hunk 4 above)
Comment (melk0sha) [reply src/components/shared/actionMenu/actionMenu.jsx]: Fixed

File: src/components/shared/input/input.jsx (hunk 5 above)
Comment (melk0sha) [reply src/components/shared/input/input.jsx]: Here we can pass any props that we could pass to input tag.
Anyway I marked this 'props' as object type.

File: src/api/__mock__/movies.json (hunk 6 above)
Comment (melk0sha) [reply src/api/__mock__/movies.json]: Fixed. Now we have 'genreIds' with ids of genres instead of whole genres objects.

**Review by cate-blint — APPROVED** [review]

**Pull request merged**